protocol/xmrmaker/message_handler.go

```diff
@@ -184,39 +184,14 @@ func (s *swapState) handleNotifyETHLocked(msg *message.NotifyETHLocked) (net.Mes
 
 	go func() {
 		until := time.Until(s.t0)
-
 		log.Debugf("time until t0: %vs", until.Seconds())
 
 		select {
 		case <-s.ctx.Done():
 			return
+		// TODO: Document why we add one second
 		case <-time.After(until + time.Second):
-			s.Lock()
-			defer s.Unlock()
-
-			if !s.info.Status().IsOngoing() {
-				// swap was already completed, just return
-				return
-			}
-
-			// we can now call Claim()
-			txHash, err := s.claimFunds()
-			if err != nil {
-				log.Errorf("failed to claim: err=%s", err)
-				// TODO: retry claim, depending on error
-				_ = s.exit()
```

Comment:
I'm just realizing with the diff view of the PR, that this `xmrmaker` version of `swapState` is calling `exit` with a lower case `e`, while the `xmrtaker` version was calling `Exit` with a capital `E`.  The lower-case version does not grab the lock, so this `xmrmaker` version was actually safe.

I would still break the 2 timeout handlers into separate functions the way this PR does to make the code readable though.